app/models/link.rb

```diff
@@ -1,4 +1,6 @@
 class Link < ActiveRecord::Base
+  after_update :reset_time_and_status, if: :url_changed?
```

Comment:
Looks like @klssmith @davidbasalla and @emmabeynon are going going to implement a presenter to take care of this. They may well refactor the above at the same time?